docs/release-notes/mongoid-8.0.txt

```diff
@@ -278,6 +278,37 @@ as well as ActiveRecord-compatible ``previously_new_record?`` and
   user.previously_persisted?   # => true
 
 
+Unknown Field Type Symbols/Strings Prohibited
+---------------------------------------------
+
+Mongoid 8 prohibits using symbols and strings as field types when these symbols
+and strings do not map to a known type. Previously such usage would create a
+field of type ``Object``.
+
+Mongoid 8 behavior:
+
+.. code-block:: ruby
+
+  class User
+    include Mongoid::Document
+
+    field :name, type: :bogus
+    # => raises Mongoid::Errors::InvalidFieldType
+  end
+
+Mongoid 7 behavior:
+
+.. code-block:: ruby
+
+  class User
+    include Mongoid::Document
+
+    field :name, type: :bogus
+    # Equivalent to:
+    field :name
```

Comment:
Is this true? I thought if you leave off the type then it's a "dynamic" field, not that it has type Object.

Reply:
It's true.
https://github.com/mongodb/mongoid/blob/master/lib/mongoid/fields.rb#L748

Reply:
Dynamic fields are those that aren't explicitly defined as fields - https://www.mongodb.com/docs/mongoid/master/reference/fields/#dynamic-fields